File: output/GPT3.5-Turbo/xCodeEval_4000/C++/Python/04d2416a6e1a2b507447d1da12867e39.py
t = int(input())

def win(a, b):
    if a % 2:
        cnt = 0
        while b:
            cnt += b % a
            b //= a
        return cnt % 2 == 0
    else:
        return b % (a + 1) % 2 == 0

def ss(a, b):
    if not a:
        return False
    if not ss(b % a, a):
        return True
    return win(a, b // a)

for _ in range(t):
    a, b = map(int, input().split())
    if ss(min(a, b), max(a, b)):
        print("First")
    else:
        print("Second")

File: output/GPT3.5-Turbo/xCodeEval_4000/C++/Go/04d2416a6e1a2b507447d1da12867e39.go
package main

import (
	"fmt"
)

func win(a, b int64) bool {
	if a%2 != 0 {
		var cnt int64 = 0
		for b != 0 {
			cnt += b % a
			b /= a
		}
		return cnt%2 == 0
	} else {
		return b%(a+1)%2 == 0
	}
}

func ss(a, b int64) bool {
	if a == 0 {
		return false
	}
	if !ss(b%a, a) {
		return true
	}
	return win(a, b/a)
}

func main() {
	var t int
	fmt.Scan(&t)
	for t > 0 {
		var a, b int64
		fmt.Scan(&a, &b)
		if ss(min(a, b), max(a, b)) {
			fmt.Println("First")
		} else {
			fmt.Println("Second")
		}
		t--
	}
}
